src/Microsoft.ML.Api/MapTransform.cs

```diff
@@ -2,319 +2,204 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using Microsoft.ML.Core.Data;
 using Microsoft.ML.Runtime.Data;
 using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.Runtime.Model;
 using System;
-using System.IO;
+using System.Linq;
 
 namespace Microsoft.ML.Runtime.Api
 {
     /// <summary>
     /// This transform generates additional columns to the provided <see cref="IDataView"/>.
     /// It doesn't change the number of rows, and can be seen as a result of application of the user's function
     /// to every row of the input data.
-    /// Similarly to the existing <see cref="IDataTransform"/>'s, this object can be treated as both the 'transformation' algorithm
-    /// (which can be then applied to different data by calling <see cref="ApplyToData"/>), and the transformed data (which can
-    /// be enumerated upon by calling <c>GetRowCursor</c> or <c>AsCursorable{TRow}</c>).
     /// </summary>
     /// <typeparam name="TSrc">The type that describes what 'source' columns are consumed from the input <see cref="IDataView"/>.</typeparam>
     /// <typeparam name="TDst">The type that describes what new columns are added by this transform.</typeparam>
-    internal sealed class MapTransform<TSrc, TDst> : LambdaTransformBase, ITransformTemplate, IRowToRowMapper
+    public sealed class CustomMappingTransformer<TSrc, TDst> : ITransformer, ICanSaveModel
```

Comment:
Can the file name be renamed too? That way people can find this class when looking through the file system.

Reply:
Will do after all other comments are resolved

---
In reply to: [229413212](https://github.com/dotnet/machinelearning/pull/1406#discussion_r229413212) [](ancestors = 229413212)